buf/validate/internal/message_rules.cc

```diff
@@ -75,10 +79,16 @@ Rules NewMessageRules(
     const google::protobuf::FieldDescriptor* field = descriptor->field(i);
     if (!field->options().HasExtension(buf::validate::field)) {
       continue;
+    }    
+    auto fieldLvl = std::ref(field->options().GetExtension(buf::validate::field));
+    if (!fieldLvl.get().has_ignore() && allMsgOneofs.count(field->name()) > 0) {      
+      FieldRules fieldLvlOvr;
+      fieldLvlOvr.CopyFrom(fieldLvl.get());
+      fieldLvlOvr.set_ignore(IGNORE_IF_UNPOPULATED);
+      fieldLvl = fieldLvlOvr;
```

Comment:
This seems to me like it would rebind the std::ref to a pointer to `fieldLvlOvr` whose lifetime ends after this scope. This may actually work but I think it is undefined behavior because the stack space is not guaranteed to be clobbered before it is read/copied. Maybe I am wrong?

I think the canonical way to do this safely is to allocate a new `FieldRules` in the current arena.